### PR TITLE
Add support CFN templates that are located in S3

### PR DESCRIPTION
**Pull request by mrterry**

### CODE REVIEW

File: kingpin/actors/aws/cloudformation.py [hunk 1]
```diff
@@ -1230,3 +1238,24 @@ def _execute(self):
...
+
+
+def parse_s3_url(url):
+    """ Gets bucket name, region, and key from url
+
+    returns bucket name, region, key
+    """
+    # virtual host style
+    # https://bucket-name.s3-Region.amazonaws.com/key-name
+    match = re.search(r'^https?://(.+)\.s3\.([^.]+)\.amazonaws\.com/(.+)', url)
```
Comment (LaikaN57) [inline kingpin/actors/aws/cloudformation.py:1250]: 1. Make the regex pattern a global so we don't have to re-compile it.
2. Use non-greedy matching for bucket (avoids bad key names) and use named groups (easier to see what we use after we match) like: `^https?://(?P<bucket>.+?)\.s3\.(?P<region>[^.]+)\.amazonaws\.com/(?P<key>.+)$`

File: kingpin/actors/aws/cloudformation.py [hunk 2]
```diff
@@ -1230,3 +1238,24 @@ def _execute(self):
...
+    if match:
+        return match.group(1), match.group(2), match.group(3)
```
Comment (LaikaN57) [inline kingpin/actors/aws/cloudformation.py:1252]: Use named groups.

File: kingpin/actors/aws/cloudformation.py [hunk 3]
```diff
@@ -1230,3 +1238,24 @@ def _execute(self):
...
+
+    # path style request
+    # http://s3.region.amazonaws.com/bucket-name/key-name
+    regex = r'^https?://s3\.([^.]+)\.amazonaws\.com/([^\/]+)/(.+)'
```
Comment (LaikaN57) [inline kingpin/actors/aws/cloudformation.py:1256]: `^https?://s3\.(?P<region>[^.]+)\.amazonaws\.com/(?P<bucket>[^/]+)/(?P<key>.+)$`

File: kingpin/actors/aws/cloudformation.py [hunk 4]
```diff
@@ -1230,3 +1238,24 @@ def _execute(self):
...
+    match = re.search(regex, url)
+    if match:
+        return match.group(2), match.group(1), match.group(3)
+
+    raise Exception('Cannot parse url')
```
Comment (LaikaN57) [inline kingpin/actors/aws/cloudformation.py:1261]: Maybe add native s3 style as well? `^s3://(?P<bucket>[^/]+)/(?P<key>.+)$`

**Comment by mrterry** [comment]
@LaikaN57 Thanks for the review!  I've take you suggestions into https://github.com/Nextdoor/kingpin/pull/508 which is the PR that is going to move forward.

**Pull request closed without merging**